common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolverSuite.java

```diff
@@ -146,12 +146,25 @@ public void jsonSerializationOfExecutorRegistration() throws IOException {
 
   @Test
   public void testNormalizeAndInternPathname() {
-    assertPathsMatch("/foo", "bar", "baz", "/foo/bar/baz");
-    assertPathsMatch("//foo/", "bar/", "//baz", "/foo/bar/baz");
-    assertPathsMatch("foo", "bar", "baz///", "foo/bar/baz");
-    assertPathsMatch("/foo/", "/bar//", "/baz", "/foo/bar/baz");
-    assertPathsMatch("/", "", "", "/");
-    assertPathsMatch("/", "/", "/", "/");
+    assertPathsMatch("/foo", "bar", "baz",
```

Comment:
Here is your comment: https://github.com/apache/spark/blob/8cd8bd1708fd492db7583bfe3f0fcfe05fb8bb75/common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExecutorDiskUtils.java#L68

Should not we use the correct path separator for testing too? What I mean is starting a path with `/` on windows 
 (like `/foo` ) is invalid in the first place.

Reply:
Not exactly. 
`\Program Files\Custom Utilities\StringFinder.exe` means an absolute path from the root of the current drive. 
Ref: https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats

Reply:
`/` != `\`
I talked about `/`  and you example is `\`.

Reply:
`/` as sep is acceptable on Windows. `\a\b\c` and `/a/b/c` both are valid and same path on Windows.

Reply:
Then your comment is invalid, right?

Reply:
What I means is: the purpose of this method is to `normalize` the path, even `/` and `\` both are acceptable sep on Windows, the `normalized` sep should be `\`. Just like `/a//b` and `/a/b` are both valid and same path on Unix, but the `normalized` expression is `/a/b`

Reply:
Sorry in this case I cannot see value in the comment. If it is not a requirement for the client of this method but a description of what the method does then I suggest to change it to:

```
    * On Windows, separator "/" is used instead of "\". 
```

Reply:
I think the right expression should be: On Windows, separator "\\" is used instead of "/".

Reply:
Sure :)